### PR TITLE
Release PR for public release r1.2 (M4)

### LINKED ISSUE #352: Create PR for public release r1.2 (M4)
**Expected action**

Create the release PR for r1.2 with the target versions for Fall24 meta-release (M4)

**Additional context**

See checklist in https://lists.camaraproject.org/g/wg-release-management/topic/meta_release_fall24_please/108038111

### PR DESCRIPTION
**Pull request by hdamker**
#### What type of PR is this?

Add one of the following kinds:
* documentation
* subproject management

#### What this PR does / why we need it:

Create the public release r1.2, see #352 

* API Readiness Checklists and CHANGELOG are anticipating the merge of
  * #349
  * #354 
  * #356 
These PRs have to be merged before the release PR!

#### Which issue(s) this PR fixes:

Fixes #352

#### Changelog input

```
 Updates of file versions, CHANGELOG and README for public release

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,31 +22,32 @@ Repository to describe, develop, document and test the QualityOnDemand API famil
...
-  * qod-provisioning v0.1.0-rc.1 (new, first initial version)  
-  [[YAML]](https://github.com/camaraproject/QualityOnDemand/blob/r1.1/code/API_definitions/qod-provisioning.yaml)
-  [[View it on ReDoc]](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/QualityOnDemand/r1.1/code/API_definitions/qod-provisioning.yaml&nocors)
-  [[View it on Swagger Editor]](https://editor.swagger.io/?url=https://raw.githubusercontent.com/camaraproject/QualityOnDemand/r1.1/code/API_definitions/qod-provisioning.yaml)
+  * qod-provisioning v0.1.0 (new, first initial version)  
+  [[YAML]](https://github.com/camaraproject/QualityOnDemand/blob/r1.2/code/API_definitions/qod-provisioning.yaml)
+  [[View it on ReDoc]](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/QualityOnDemand/r1.2/code/API_definitions/qod-provisioning.yaml&nocors)
+  [[View it on Swagger Editor]](https://editor.swagger.io/?url=https://raw.githubusercontent.com/camaraproject/QualityOnDemand/r1.2/code/API_definitions/qod-provisioning.yaml)
+
+* Previous releases and pre-releases are available here: https://github.com/camaraproject/QualityOnDemand/releases/latest
```
Comment (hdamker) [inline README.md:47]:
```suggestion
* Previous releases and pre-releases are available here: https://github.com/camaraproject/QualityOnDemand/releases
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,10 +13,108 @@
...
+
+### Fixed
+* n/a
+
+### Removed
+* n/a
+
+**Full Changelog**: https://github.com/camaraproject/QualityOnDemand/compare/v0.10.1...r1.2
+
 # r1.1
```
Comment (jlurien) [inline CHANGELOG.md:118]: Do we keep r1.1 in the changelog or should we remove the pre-release once we have the public release? It is really duplicating the same info and it does not add any real value, IMO.

Comment (hdamker) [reply CHANGELOG.md:118]: See discussion in https://github.com/camaraproject/DeviceLocation/pull/252#issuecomment-2312402603

**Comment by hdamker** [comment]
Added #356 proactively to CHANGELOG and updated the PR description.

**Comment by hdamker** [comment]
All PRs for v0.11.0 so far closed, adding @camaraproject/release-management_maintainers as reviewer.

**Comment by hdamker** [comment]
@jlurien @RandyLevensalor @eric-murray @tlohmar: I've addressed #340 here with https://github.com/camaraproject/QualityOnDemand/pull/353/commits/e7bd70188974562853d3d96132ca54585b560c6b. Please have a look if that is ok for you.

**Comment by tanjadegroot** [comment]
I do not see the updated User Story file in this PR, but I have a comment on the Summary section: 
"As an application developer belonging to an enterprise, I want to request (using my application server/backend service), stable latency and/or stable throughput for a specified App-Flow from a Communication Service Provider (CSP), so that I can ensure better quality of experience for our end users including under network congestion"

I would propose to drop the "from a Communication Service Provider (CSP)" as the app can also talk to an Aggregator. 
Or would it be "for a specified App-Flow from a given device connected to the application through the network, so that ..." ?

Also suggest to drop the word "our" from "our end users".

**Comment by tanjadegroot** [comment]
I think you can remove the LICENCE.APACHE.2.0 file from the code/Test_definitions folder as it is already present in the main folder.

**Review by tanjadegroot — COMMENTED** [review]
M4 release management review done: A few minor comments only 
please check and then LGTM

File: README.md [hunk 3]
```diff
@@ -22,31 +22,32 @@ Repository to describe, develop, document and test the QualityOnDemand API famil
...
-  [[View it on ReDoc]](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/QualityOnDemand/r1.1/code/API_definitions/qod-provisioning.yaml&nocors)
-  [[View it on Swagger Editor]](https://editor.swagger.io/?url=https://raw.githubusercontent.com/camaraproject/QualityOnDemand/r1.1/code/API_definitions/qod-provisioning.yaml)
+  * qod-provisioning v0.1.0 (new, first initial version)  
+  [[YAML]](https://github.com/camaraproject/QualityOnDemand/blob/r1.2/code/API_definitions/qod-provisioning.yaml)
+  [[View it on ReDoc]](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/QualityOnDemand/r1.2/code/API_definitions/qod-provisioning.yaml&nocors)
+  [[View it on Swagger Editor]](https://editor.swagger.io/?url=https://raw.githubusercontent.com/camaraproject/QualityOnDemand/r1.2/code/API_definitions/qod-provisioning.yaml)
+
+* Previous releases and pre-releases are available here: https://github.com/camaraproject/QualityOnDemand/releases
+* For changes see [CHANGELOG.md](https://github.com/camaraproject/QualityOnDemand/blob/main/CHANGELOG.md)
```
[the remark this answers is not in the document]
Comment (hdamker) [reply README.md:48]: By intention to main - which has always the latest changes and the history.

File: documentation/API_documentation/quality-on-demand-API-Readiness-Checklist.md [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
+|  6 | User stories                                 |   O   |         O         |    O    |    M   | Y    | [/documentation/API_documentation/QoD_User_Story.md](/documentation/API_documentation/QoD_User_Story.md) |
+|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   | Y    | [/code/Test_definitions](/code/Test_definitions) |
+|  8 | Enhanced API test cases & documentation      |   O   |         O         |    O    |    M   | N    |      |
 |  9 | Test result statement                        |   O   |         O         |    O    |    M   | N    |      |
 | 10 | API release numbering convention applied     |   M   |         M         |    M    |    M   | Y    |      |
-| 11 | Change log updated                           |   M   |         M         |    M    |    M   | Y    | /CHANGELOG.md |
+| 11 | Change log updated                           |   M   |         M         |    M    |    M   | Y    | [/CHANGELOG.md](/CHANGELOG.md) |
 | 12 | Previous public release was certified        |   O   |         O         |    O    |    M   | ?    |      |
```
Comment (hdamker) [inline documentation/API_documentation/quality-on-demand-API-Readiness-Checklist.md:18]:
```suggestion
| 12 | Previous public release was certified        |   O   |         O         |    O    |    M   |   Y    |   *   |

* GSMA certified implementation by China Unicom, multiple other implementation by operators  (source: https://www.open-gateway.com/operators-map as of 2024-08-16) 
```

File: code/API_definitions/quality-on-demand.yaml
[the remark this answers is not in the document]
Comment (hdamker) [reply code/API_definitions/quality-on-demand.yaml:1]: An explanation for developers would make sense (note to myself to create an issue for it). 
The reason for this construct is the Design Guideline in https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#post-or-get-for-transferring-sensitive-or-complex-data

[the remark this answers is not in the document]
Comment (hdamker) [reply code/API_definitions/quality-on-demand.yaml:1]: DeviceIPv4Address is a complex object due the fact that a single (public or private) IPv4 address is not enough to identify a device in an operator network.
ApplicationServerIPv4Address (and same for IPv6) is not a single IP address but allows subnets. There is no common type for that.

File: README.md [hunk 5]
```diff
@@ -22,31 +22,32 @@ Repository to describe, develop, document and test the QualityOnDemand API famil
 * Started: October 2021
 * Location: virtually
```
[the remark this answers is not in the document]
Comment (hdamker) [reply README.md]: Done in https://github.com/camaraproject/QualityOnDemand/pull/353/commits/cdf39329764126d2f2d65e06887a48967c31e710

File: code/API_definitions/quality-on-demand.yaml
[the remark this answers is not in the document]
Comment (hdamker) [reply code/API_definitions/quality-on-demand.yaml:1]: It is a POST operation which does not create a resource, so it should be a verb. But I agree that "extend-duration" would have been more clear.

**Comment by hdamker** [comment]
@tanjadegroot Thanks a lot for the intensive review. See my answer on your comments. Please have a view if ok for you. I have also deleted the duplicated license file (I recently added the file at the right place but have overseen that there was one in /code). For the User Story I don't see the changes as urgent and have kept it untouched (but I merged `main` into the PR branch, so that now the User Story is part of it).

@eric-murray @jlurien @RandyLevensalor please check the latest changes and approve the PR if ok for you.

File: code/API_definitions/quality-on-demand.yaml
[the remark this answers is not in the document]
Comment (hdamker) [reply code/API_definitions/quality-on-demand.yaml:1]: Issue #359 created, to be addressed in next release

**Comment by hdamker** [comment]
@jlurien @eric-murray @RandyLevensalor I've addressed all review comments from Tanja, so from Release Management there is green light. But I can't merge the release PR without a Codeowner review.

**Review by RandyLevensalor — APPROVED** [review]
LGTM

@hdamker thanks for all of the updates and work on this!!

File: code/API_definitions/quality-on-demand.yaml
[the remark this answers is not in the document]
Comment (tanjadegroot) [reply code/API_definitions/quality-on-demand.yaml:1]: ah, OK, I did not know that design rule and thought POST is always about creating a resource

**Review by tanjadegroot — APPROVED** [review]

**Comment by tanjadegroot** [comment]
> @tanjadegroot Thanks a lot for the intensive review. See my answer on your comments. Please have a view if ok for you. I have also deleted the duplicated license file (I recently added the file at the right place but have overseen that there was one in /code). For the User Story I don't see the changes as urgent and have kept it untouched (but I merged `main` into the PR branch, so that now the User Story is part of it).

Review done by RM: LGTM

**Pull request merged**